daml-assistant/src/DA/Daml/Assistant/Install/Path.hs

```diff
@@ -77,7 +80,36 @@ regQueryStringValue key valueName =
                 c_RegQueryValueEx p_key c_valueName nullPtr nullPtr c_value p_valueLen
             peekTString (castPtr c_value)
 #else
-updatePath _ output targetPath = suggestAddToPath output targetPath
+updatePath installOpts output targetPath
+  | SetPath b <- iSetPath installOpts
+  , not b = suggestAddToPath output targetPath
+  | otherwise = do
+    searchPaths <- map dropTrailingPathSeparator <$> getSearchPath
+    unless (targetPath `elem` searchPaths) $ do
+      mbShellPath <- lookupEnv "SHELL"
+      let mbShellConfig = do
+            shellPath <- mbShellPath
+            shellConfig (takeFileName shellPath) targetPath
+      case mbShellConfig of
+        Nothing -> output setPathManualMsg
+        Just (configFile, cmd) -> do
+          home <- getHomeDirectory
+          let cfgFile = home </> configFile
+          cfgFileExists <- doesFileExist cfgFile
+          if cfgFileExists
+            then do
+              content <- readFile cfgFile
+              unless (cmd `elem` lines content) $ do
+                stdinIsTerminal <- hIsTerminalDevice stdin
+                stdoutIsTerminal <- hIsTerminalDevice stdout
+                if stdinIsTerminal && stdoutIsTerminal
+                  then do
+                    answer <- prompt "Add DAML Assistant executable to your PATH?" "Yes" ["No"]
```

Comment:
Maybe it's good to make the file we're going to change explicit so expert users can press "No" if they want their path somewhere else?
```suggestion
                    answer <- prompt ("Add DAML Assistant executable to your PATH (in " ++ configFile ++ ")?") "Yes" ["No"]
```

Reply:
I'm fine with that. Could also be scary for people that have no idea of the existence of that file.